server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -677,6 +677,13 @@ public List<RepositoryStatsSnapshot> repositoriesStats() {
         return repositoriesStats;
     }
 
+    public RepositoriesStats getRepositoriesThrottlingStats() {
+        RepositoriesStats.Builder b = RepositoriesStats.builder();
```

Comment:
Do we need a builder here? I don't think it's adding much over just creating the map directly.

Reply:
I've been adding builder here to have some extensibility in case we decide to add more stats.

Reply:
Adding a builder later is an easy refactoring to do. Let's keep it simple for now and revisit this if/when we need the extra indirection.

Reply:
OK, simplified